src/store/epics/accounts.ts

```diff
@@ -1,40 +1,41 @@
+import { merge, of } from 'rxjs'
+import { fromPromise } from 'rxjs/observable/fromPromise'
+import { catchError, filter, map, pluck, switchMap, withLatestFrom } from 'rxjs/operators'
+import { isActionOf } from 'typesafe-actions'
+
 import {
   accountIsLoading,
   accountLoadFailure,
   accountLoadRequest,
   accountLoadSuccess,
   loadAccountTransactions,
 } from '@actions'
-import { RootEpic } from '@store'
-import { merge, of } from 'rxjs'
-import { fromPromise } from 'rxjs/observable/fromPromise'
-import { catchError, delay, filter, map, mergeMap, withLatestFrom } from 'rxjs/operators'
-import { isActionOf } from 'typesafe-actions'
+import { RootAction, RootEpic } from '@store'
 
-export const loadAccount$: RootEpic = (action$, state$, { loadAccount }) => {
+export const loadAccount$: RootEpic = (action$, state$, { getCurrentConnection, loadAccount, withPolling }) => {
   const accountLoadRequest$ = action$.pipe(
     filter(isActionOf(accountLoadRequest)),
-    map(({payload}) => payload),
   )
 
   const accountIsLoading$ = accountLoadRequest$.pipe(
     map(() => accountIsLoading()),
   )
 
   const accountLoad$ = accountLoadRequest$.pipe(
-    delay(500),
+    withPolling(350, 20000),
+    pluck<RootAction, string>('payload'),
     withLatestFrom(state$),
-    mergeMap(
-      ([publicKey, state]) => fromPromise(loadAccount(publicKey, state.connections.currentConnection))
-        .pipe(
-          map(accountLoadSuccess),
-          catchError((err) => of(accountLoadFailure(err))),
+    switchMap(([ publicKey, state ]) =>
+      fromPromise(loadAccount(publicKey, getCurrentConnection(state))).pipe(
+        map(accountLoadSuccess),
+        catchError((err) => of(accountLoadFailure(err))),
       ),
-    ),
+    )
```

Comment:
You could also write this as:
```ts
accountLoadRequest$.pipe(
pluck('payload'),
interval(500),
startWith(0),
withLatestFrom(state$),
switchMap(),
takeUntil(accountLoadRequest$),
)
```
Maybe... untested

Reply:
```ts
  const invalidatePoll$ = merge(
    accountLoadRequest$,
    state$.pipe(
      pluck<RootState, WalletView>('view', 'walletView'),
      filter(view => view !== WalletView.dashboard),
    ),
    state$.pipe(
      pluck<RootState, AccountPage>('accountPage', 'accountPage'),
      filter(page => page !== AccountPage.dashboard),
    ),
  )
  const accountLoadPoll$ = accountLoadRequest$.pipe(
    switchMap(action =>
      interval(1000).pipe(
        startWith(0),
        withLatestFrom(state$),
        switchMap(([_, state]) =>
          fromPromise(loadAccount(action.payload, getCurrentConnection(state))).pipe(
            map(accountLoadSuccess),
            catchError(err => of(accountLoadFailure(err))),
          ),
        ),
        takeUntil(invalidatePoll$),
      ),
    ),
```
This was able to work, although currently there isn't an accountLoadRequest when we switch from Signing page to Dashboard

Reply:
I've reworked above again to skip when not on the account dashboard page. This I think is a little nicer, but maybe harder to test.
```ts
export const loadAccount$: RootEpic = (action$, state$, { getCurrentConnection, loadAccount }) => {
  const accountLoadRequest$ = action$.pipe(filter(isActionOf(accountLoadRequest)))

  const accountIsLoading$ = accountLoadRequest$.pipe(mapTo(accountIsLoading()))

  // If there are other things that would invalidate the polling, should add to here
  const invalidatePoll$ = merge(
    accountLoadRequest$,
    state$.pipe(
      pluck<RootState, WalletView>('view', 'walletView'),
      filter(view => view !== WalletView.dashboard),
    ),
  )

  const accountLoadPoll$ = accountLoadRequest$.pipe(
    mergeMap(action =>
      interval(5000).pipe(
        startWith(0),
        withLatestFrom(state$),
        // Want to skip while not focused on dashboard page
        skipWhile(([_, state]) => state.accountPage.accountPage !== AccountPage.dashboard),
        switchMap(([_, state]) =>
          fromPromise(loadAccount(action.payload, getCurrentConnection(state))).pipe(
            map(accountLoadSuccess),
            catchError(err => of(accountLoadFailure(err))),
          ),
        ),
        takeUntil(invalidatePoll$),
      ),
    ),
  )

  const loadAccountTransactions$ = accountLoadRequest$.pipe(
    pluck<RootAction, string>('payload'),
    map(loadAccountTransactions),
  )

  return merge(accountIsLoading$, accountLoadPoll$, loadAccountTransactions$)
}
```